engineyard/ey-cookbooks-stable-v6
# [TEY-108] Exclude DB instances from hosting Sidekiq workers

Description of your patch
-------------
This PR updates the sidekiq recipes (including custom sidekiq) to exclude DB instances from hosting sidekiq workers by default.

Recommended Release Notes
-------------
Excludes DB instances from hosting Sidekiq workers by default

Estimated risk
-------------
Low. Only an additional check for `is_sidekiq_instance` was added.

Components involved
-------------
sidekiq recipes (and custom sidekiq)

Dependencies
-------------
Depends on #159.

Description of testing done
-------------
On a Rails ActiveJob testing app:
1. Booted an environment (with these changes) and the following configuration:
   a. App master, App slave, DB master, DB slave, sidekiq util, and redis util instances
   c. Environment variables: `EY_REDIS_ENABLED=true` and `EY_SIDEKIQ_ENABLED=true`
2. Checked if environment boots successfully and Chef run succeeds.
3. Checked all instances if sidekiq workers are running as expected (no sidekiq workers on the DB instances)

QA Instructions
-------------
Check on a solo environment.
Check explicitly selecting DB instances for sidekiq workers.
Test a PHP application.
Test a default Ruby application.
